### PR TITLE
SGS-AMD: move to AtmosModel

### PR DESCRIPTION
**Pull request by asraero**
Anisotropic Minimum Dissipation model of Rozema et al. 
- [x] Tested with RTB, DC, DYCOMS cases (DYCOMS currently underway @2000s simtime) 

TODO: Include diffusivity according to Rozema's method, as opposed to simplified approximation via turbulent Prandtl number (eddy viscosity still computed according to Rozema et al.(2015)). Can be introduced as a separate PR if necessary. Future PR includes the results from the 2018 Abkar et al. paper
```
Reference: 
(Rozema, W., Bae, H. J., Moin, P., & Verstappen, R. (2015). Minimum-dissipation models for large-eddy simulation. Physics of Fluids, 27(8), [085107]. https://doi.org/10.1063/1.4928700) 
```

### CODE REVIEW

**Comment by asraero** [comment]
bors try

**Comment by glwagner** [comment]
@asraero, just wanted to share a couple of thoughts about AMD:

- Abkar and Moin (2017) have extended the AMD model to account for the affects of buoyancy: 

https://link.springer.com/article/10.1007/s10546-017-0288-4

We haven't tested the effect of the buoyancy related term extensively yet in `Oceananigans.jl`, so I'm not sure how much of a difference it makes. 

- Verstappen (2018) proposes a modification to AMD that scales both the gradients of tracers and velocity fields, as well as the velocity fields themselves:

https://www.sciencedirect.com/science/article/abs/pii/S0045793016304005

The resulting model is slightly different than that proposed by Rozema et al. (2015). This model was also implemented and tested by Vreugdenhil and Taylor (2018):

https://aip.scitation.org/doi/abs/10.1063/1.5037039

In conversations with V&T, it seems that the "Verstappen" version of AMD performs better for very anisotropic grids. However, the differences have not been quantified. In `Oceananigans.jl`, we have implemented both the "Rozema" AMD and the "Verstappen" AMD; because of the conversations with V&T, we currently use the Verstappen AMD as our default.

Finally, I think the model for tracer subgrid diffusivity is interesting and worth implementing.

**Comment by asraero** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by simonbyrne** [comment]
bors try

**Comment by asraero** [comment]
bors try

**Comment by kpamnany** [comment]
> ## try
> # Build failed
> * [ci/slurmci](https://gist.github.com/833a5f8f3814479b775e1f52a1e0356d)

This is a timeout due to the change to `density_current-model.jl`; it is taking longer than 30 minutes to run on both CPU and GPU. We can change the requested time for the run if that's what you intend @asraero ?

**Comment by asraero** [comment]
bors try

File: src/Atmos/Model/boundaryconditions.jl [hunk 1]
```diff
@@ -226,3 +226,55 @@ function atmos_boundary_state!(::CentralNumericalFluxDiffusive, bc::DYCOMS_BC,
...
+    FT = eltype(stateP)
+    ρP  = stateM.ρ
+    # Weak Boundary Condition Imposition
+    # Prescribe no-slip wall (Dirichlet b.c. for wall velocity)
+    # Note that with the default resolution this results in an underresolved near-wall layer
+    # In the limit of Δ → 0, the exact boundary values are recovered at the "M" or minus side. 
+    # The weak enforcement of plus side states ensures that the boundary fluxes are consistently calculated.
+    UP  = FT(0)
+    VP  = FT(0) 
+    WP  = FT(0)
```
Comment (simonbyrne) [inline src/Atmos/Model/boundaryconditions.jl]:
```suggestion
    ρuP  = SVector{3,FT}(0,0,0) 
```

**Pull request merged**